NSS-Therapeutic-Raccoons/BangazonSite
# Rb 7 delete order product

## Link to Ticket
Closes # [7](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/7)
Closes # [43](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/43)

## Description of Proposed Changes
- Add a "remove" button for products in the cart that removes the order product from the cart. 
- Fix bug that allows a negative quantity to be inputed for creating products.


## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout rb-7-deleteOrderProduct
```

Areas of the ticket to examine and ensure that everything works properly
* Login and click view cart.
* verify that an active order exists, and has order products in it.
* If it does not, search for a product, click add to order and you will be redirected to an order with that product in it.
* Return to that products details to verify that the quantity went down by one. Go back to cart.
* verify that remove button is present on the same row as the product.
* click the button, and verify that the order product is removed from the order.
* if it was the only product, verify that you are returned to a screen that shows that the cart is empty.
* Return to that products details and verify that the quantity increased by one.
* Go to Sell Something on the NavBar, and attempt to create a product with a negative quantity. Verify that it will not let you create unless quantity greater than 0.

## Impacted Areas in Application

List general components of the application that this PR will affect:

* `ProductsController` to add the `RemoveCartProduct` Method.
* `Models/OrderViewModels/OrderLineItem.cs` had an OrderProductId added to it.
* `ProductsController` Method Purchase updated to add OrderProductId to OrderLineItems 
* `Views/Orders/Details.cshtml` to rework the display of order products and add the remove button.

## Mentions @username

@figamus
@ricky-bruner
@markhale94
@taylorlgulley
@askingalot

## Definition of Done

1. The project must be fully documented. This includes the following:
	* Comments for classes and methods added
	* Comments for models, controllers, and views added
2. The testing steps must completely work per ticket requirements.
3. Andy must smile approvingly and smirk with satisfaction following the smile.
